src/libsync/propagateupload.cpp

```diff
@@ -268,6 +270,8 @@ void PropagateUploadFileQNAM::slotComputeTransmissionChecksum(const QByteArray&
 
     connect(computeChecksum, SIGNAL(done(QByteArray,QByteArray)),
             SLOT(slotStartUpload(QByteArray,QByteArray)));
+    connect(computeChecksum, SIGNAL(done(QByteArray,QByteArray)),
+            computeChecksum, SLOT(deleteLater()));
     const QString filePath = _propagator->getFilePath(_item->_file);
```

Comment:
All those are optimizations for within propagation, we don't have real memory leaks.
@dragotin You can mix deleteLater, call them multiple times and have them mixed with QObject parenting without problem as long as you don't have a dangling pointer (there isn't any in this case).